XCUITests/FxScreenGraph.swift

```diff
@@ -0,0 +1,181 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import XCTest
+
+let SettingsScreen = "SettingsScreen"
+let TabTray = "TabTray"
+let TabTrayMenu = "TabTrayMenu"
+let BrowserTab = "BrowserTab"
+let BrowserTabMenu = "BrowserTabMenu"
+let BrowserTabMenu2 = "BrowserTabMenu2"
+let FirstRun = "OptionalFirstRun"
+let HomePageSettings = "HomePageSettings"
+let PasscodeSettings = "PasscodeSettings"
+let PasscodeIntervalSettings = "PasscodeIntervalSettings"
+let SearchSettings = "SearchSettings"
+let NewTabSettings = "NewTabSettings"
+let ClearPrivateDataSettings = "ClearPrivateDataSettings"
+let LoginsSettings = "LoginsSettings"
+let OpenWithSettings = "OpenWithSettings"
+let NewTabScreen = "NewTabScreen"
+let NewTabMenu = "NewTabMenu"
+
+let allSettingsScreens = [
+    SettingsScreen,
+    HomePageSettings,
+    PasscodeSettings,
+    SearchSettings,
+    NewTabSettings,
+    ClearPrivateDataSettings,
+    LoginsSettings,
+    OpenWithSettings,
+]
+
+func createScreenGraph(app: XCUIApplication, url: String = "https://www.mozilla.org/en-US/book/") -> ScreenGraph {
+    let map = ScreenGraph(app)
+
+    map.createScene(FirstRun) { scene in
+        scene.gesture(to: NewTabScreen) {
+            let firstRunUI = app.buttons["Start Browsing"]
+            if (firstRunUI.exists) {
```

Comment:
I've found three classes of error:

 1. if you have tried to navigate to a screen that is not possible to get to from your current position. In this case, somehow there are unreachable nodes in the graph. This is a fault with the the declared graph not reflecting the actual application graph, most likely caused by `FxScreenGraph` being incomplete. In this case the error is reported at the `navigator.goto(UnreachableScreen)` line in your test.
 2. The `XCUIApplication` is unable to see the given `UIView`. Where possible, the error is reported in `FxScreenGraph`.
 3. The application state is such that a screen cannot be legitimately reached. e.g. the `PasscodeIntervalSettings` screen can't be reached unless a passcode is enabled. Only then is the cell enabled. At this point, the navigator will tap on the table cell, but it's disabled, so nothing will happen, and the navigator will get out of sync with the application. These will manifest as error type (2). 

Distinguishing between type 2 and type 3 errors _should_ be fairly simple, but I didn't get round to reliably doing it without changing the element itself.